src/main/java/de/themorpheus/edu/taskservice/controller/solution/SolutionController.java

```diff
@@ -6,12 +6,21 @@
 import de.themorpheus.edu.taskservice.database.repository.solution.SolutionRepository;
 import de.themorpheus.edu.taskservice.util.ControllerResult;
 import de.themorpheus.edu.taskservice.util.Error;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.TypeFilter;
 import org.springframework.stereotype.Controller;
 
 @Controller
 public class SolutionController {
 
+	private static List<Object> solutionControllers = new ArrayList<>();
```

Comment:
why static?

Reply:
and why object when u know its of type "Solution" (interface)

Reply:
static because there are multiple SolutionController instances